Automattic/sensei
# Add course list filter block

Implements https://github.com/Automattic/sensei/issues/5559

### Changes proposed in this Pull Request

* Added a filter block for filtering the courses shown inside Course List block. Filtering with categories is implemented. Filtering by activity and featured will be done as part of other cards.

### Testing instructions

- Create some courses
- Create some course categories
- Assign some courses in some categories
- Add a Course List block on any page
- Add the Course List Filter block inside the Course List block
- Save and go to frontend
- Make sure it is rendering properly
- Try filtering by changing the dropdown value to different course categories

### Screenshot / Video

https://user-images.githubusercontent.com/6820724/187975237-d42479f1-2b46-492c-b5b7-c6fdaded4bdc.mov
